src/protocols/coda_transition_frontier.ml

```diff
@@ -332,14 +308,56 @@ module type Transition_frontier_controller_intf = sig
     -> network:network
     -> time_controller:time_controller
     -> frontier:transition_frontier
-    -> transition_reader:( [ `Transition of external_transition
+    -> transition_reader:( [ `Transition of external_transition_verified
                                             Envelope.Incoming.t ]
                          * [`Time_received of time] )
                          Reader.t
     -> clear_reader:[`Clear] Reader.t
     -> (external_transition_verified, state_hash) With_hash.t Reader.t
 end
 
+module type Protocol_state_validator_intf = sig
+  type time
+
+  type state_hash
+
+  type external_transition
+
+  type external_transition_verified
+
+  val validate_proof :
```

Comment:
it's a bit scary to make this public since it gives you back a verified external transition without checking consensus state

Reply:
see my other comment about making `external_transition_proof_verified`